cmd/mongodb/main.go

```diff
@@ -0,0 +1,139 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"os"
+	"os/signal"
+	"syscall"
+
+	kitprometheus "github.com/go-kit/kit/metrics/prometheus"
+	"github.com/mainflux/mainflux"
+	log "github.com/mainflux/mainflux/logger"
+	"github.com/mainflux/mainflux/writers"
+	mongodb "github.com/mainflux/mainflux/writers/mongodb"
+	"github.com/mongodb/mongo-go-driver/mongo"
+	nats "github.com/nats-io/go-nats"
+	stdprometheus "github.com/prometheus/client_golang/prometheus"
+)
+
+const (
+	name       = "mongodb-writer"
+	defNatsURL = nats.DefaultURL
+	defPort    = "8180"
+	defDBName  = "mainflux"
+	defDBHost  = "localhost"
+	defDBPort  = "27017"
+	defDBUser  = "mainflux"
+	defDBPass  = "mainflux"
+
+	envNatsURL = "MF_NATS_URL"
+	envPort    = "MF_MONGO_WRITER_PORT"
+	envDBName  = "MF_MONGO_WRITER_DB_NAME"
+	envDBHost  = "MF_MONGO_WRITER_DB_HOST"
+	envDBPort  = "MF_MONGO_WRITER_DB_PORT"
+	envDBUser  = "MF_MONGO_WRITER_DB_USER"
+	envDBPass  = "MF_MONGO_WRITER_DB_PASS"
+)
+
+type config struct {
+	NatsURL string
+	Port    string
+	DBName  string
+	DBHost  string
+	DBPort  string
+	DBUser  string
+	DBPass  string
+}
+
+func main() {
+	cfg := loadConfigs()
+	logger := log.New(os.Stdout)
+
+	nc, err := nats.Connect(cfg.NatsURL)
+	if err != nil {
+		logger.Error(fmt.Sprintf("Failed to connect to NATS: %s", err))
+		os.Exit(1)
+	}
+	defer nc.Close()
+
+	ms, err := connect("mongodb://"+cfg.DBHost+":"+cfg.DBPort, cfg.DBName)
```

Comment:
I would use `sprintf` here to form temporary mongoURL var, it will be more readable.

`mongoURL := sprintf("mongodb://%s:%s", cfg.DBHost, cfg.DBPort)`

Reply:
@anovakovic01 @dusanb94 what do you think about this ^

Reply:
@drasko I agree.